File: output/deepseek_chat/avatar/direct_translation/java/python/successful_translation/atcoder_ABC132_A.py

s = input()
count = {}
for char in s:
    if char in count:
        count[char] += 1
    else:
        count[char] = 1
if all(value == 2 for value in count.values()) and len(count) == 2:
    print("Yes")
else:
    print("No")

File: output/deepseek_r1/avatar/direct_translation/python/java/atcoder_ABC132_A.java

import java.util.*;

public class atcoder_ABC132_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String S = scanner.next();
        HashMap<Character, Integer> count = new HashMap<>();
        for (char c : S.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        if (count.size() != 2) {
            System.out.println("No");
            System.exit(0);
        }
        for (int val : count.values()) {
            if (val != 2) {
                System.out.println("No");
                System.exit(0);
            }
        }
        System.out.println("Yes");
    }
}

